src/codeu/chat/server/Model.java

```diff
@@ -148,11 +149,75 @@ public void changeAuthority(ConversationHeader conversation, Uuid targetUser, by
     userAuthority.put(targetUser, authorityByte);
   }
 
-  public boolean validateAuthority(ConversationHeader conversation, Uuid targetUser, byte authorityByte)
+  public boolean isBannedUser(ConversationHeader conversation,Uuid targetUser)
+  {
+    LinkedHashMap<Uuid, Byte> userAuthority = authority.get(conversation.id);
+    Byte authoritByte = userAuthority.get(targetUser);
+    return authoritByte != null && authoritByte.byteValue() == Controller.USER_TYPE_BANNED;
+  }
+
+
+  public boolean isBannedUser(Uuid conversation,Uuid targetUser)
+  {
+    LinkedHashMap<Uuid, Byte> userAuthority = authority.get(conversation);
+    Byte authoritByte = userAuthority.get(targetUser);
+    return authoritByte != null && authoritByte.byteValue() == Controller.USER_TYPE_BANNED;
+  }
+
+  public boolean isUser(ConversationHeader conversation,Uuid targetUser)
   {
     LinkedHashMap<Uuid, Byte> userAuthority = authority.get(conversation.id);
-    return userAuthority.get(conversation.id) != null && userAuthority.get(targetUser).byteValue() == authorityByte;
+    Byte authoritByte = userAuthority.get(targetUser);
+    if(authoritByte == null)
+    {
+      userAuthority.put(targetUser, Controller.USER_TYPE_USER);
+      return true;
+    }
+    return authoritByte.byteValue() == Controller.USER_TYPE_USER;
+  }
+
+
+  public boolean isUser(Uuid conversation,Uuid targetUser)
```

Comment:
You should change both of your isUser methods to isMember because otherwise we don't know what you mean. A good way to think about it is that just like squares are a sort of rectangle, members are a kind of user.